Note that `convert` does not handle exceptions (#30)
parsonsmatt/parsonsmatt.github.io: _posts/2017-06-21-exceptional_servant_handling.markdown

convert = Handler . ExceptT . try
```

> **NOTE**: This `convert` function is a no-op. `try`'s `e` type is instantiated to `ServerError` here, and no `IO a` actions you pass will throw `ServerError`, meanwhile any `IOException` exception thrown by IO actions will not be caught by `try`. So `convert` is basically doing nothing.

Reviewer: The point of this blog post is using `IO` exceptions for the `ServantErr`, which the above `server` does.

```suggestion
**NOTE**: This `convert` function *only handles `ServantErr`. `try`'s `e` type is instantiated to `ServantErr` here. Any `IOException` exception thrown by IO actions will not be caught by `try` here.
```

wrap it in the `NT` natural transformation newtype:

```haskell
convert :: IO :~> Handler
convert = NT . Handler . ExceptT . try
```

and use it in `enter`:

```haskell
app :: Application
app = server (Proxy :: Proxy Api) (enter convert handler)
```

> Voila! You're throwing exceptions in IO, and Servant is still doing nothing with them.

Reviewer: ```suggestion
Voila! You're throwing and catching all your exceptions in IO, and Servant is still getting the chance to handle the `ServantErr` exceptions as HTTP responses. Servant doesn't try to handle any `IO` exceptions that make it through and allows the web server to handle them.
```
